scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/testmode/NonMapBasedRecordTypesOps.scala

```diff
@@ -0,0 +1,46 @@
+package pl.touk.nussknacker.engine.testmode
+
+import pl.touk.nussknacker.engine.api.typed.typing._
+import pl.touk.nussknacker.engine.util.Implicits.RichScalaListMap
+
+import java.util.{Map => JMap}
+
+object NonMapBasedRecordTypesOps {
+
+  implicit class TypingResultExt(typingResult: TypingResult) {
+
+    def toMapBasedRecordTypes: TypingResult = typingResult match {
+      case union: TypedUnion          => Typed(union.possibleTypes.map(_.toMapBasedRecordTypes))
+      case TypedNull                  => TypedNull
+      case single: SingleTypingResult => single.toMapBasedRecordTypes
+      case unknown: Unknown           => unknown
+    }
+
+  }
+
+  private implicit class SingleTypingResultExt(single: SingleTypingResult) {
+
+    def toMapBasedRecordTypes: SingleTypingResult = single match {
```

Comment:
can we have unit test for the class which could explain what the method does?

Reply:
This method is covered by high-level tests such as `SchemedKafkaScenarioTestingSpec`, `BatchDataGenerationSpec`, but I can also add low-level tests.

Reply:
The main point of this PR was that it wasn't used in `CommonDataFormatHandler.getTestParametersDefinition` - it was only used in  `CommonTestDataFormatStubbedSourcePreparer`. I added the totally new test suite (`SchemedKafkaScenarioTestingSpec`) covering all testing endpoint logic for Avro types, including this one endpoint, using `getTestParametersDefinition` under the hood, that was broken for Avro.

I understand your point of view. I added some unit tests, a comment explaining what it does and I did some small cleanups to emphasize even more what transformations on types we do during scenario testing.